TLM/TLM/Manager/Impl/TurnOnRedManager.cs

```diff
@@ -1,233 +1,273 @@
+                               "outgoing one-way or insufficient number of outgoing segments.");
+                }
+
+                return;
+            }
+
+            bool lhd = Services.SimulationService.LeftHandDrive;
+
+            // check node
+            // note that we must not check for the `TrafficLights` flag here because the flag might not be loaded yet
+            bool nodeValid = false;
+            Services.NetService.ProcessNode(
+                nodeId,
+                (ushort _, ref NetNode node) => {
+                    nodeValid =
+                        (node.m_flags & NetNode.Flags.LevelCrossing) ==
+                        NetNode.Flags.None &&
+                        node.Info?.m_class?.m_service != ItemClass.Service.Beautification;
+                    return true;
+                });
+
+            if (!nodeValid) {
+                if (logTurnOnRed) {
+                    Log._Debug($"TurnOnRedManager.UpdateSegmentEnd({end.segmentId}, {end.startNode}): node invalid");
+                }
+
+                return;
+            }
+
+            // get left/right segments
+            ushort leftSegmentId = 0;
+            ushort rightSegmentId = 0;
+            Services.NetService.ProcessSegment(
+                end.segmentId,
+                (ushort _, ref NetSegment segment) => {
+                    segment.GetLeftAndRightSegments(
+                        nodeId,
+                        out leftSegmentId,
+                        out rightSegmentId);
+                    return true;
+                });
+
+            if (logTurnOnRed) {
+                Log._Debug(
+                    $"TurnOnRedManager.UpdateSegmentEnd({end.segmentId}, {end.startNode}): " +
+                    $"got left/right segments: {leftSegmentId}/{rightSegmentId}");
+            }
+
+            // validate left/right segments according to geometric properties
+            if (leftSegmentId != 0
+                && segmentEndManager.GetDirection(ref end, leftSegmentId) != ArrowDirection.Left)
+            {
+                if (logTurnOnRed) {
+                    Log._Debug(
+                        $"TurnOnRedManager.UpdateSegmentEnd({end.segmentId}, {end.startNode}): " +
+                        "left segment is not geometrically left");
+                }
+
+                leftSegmentId = 0;
+            }
+
+            if (rightSegmentId != 0
+                && segmentEndManager.GetDirection(ref end, rightSegmentId) != ArrowDirection.Right)
+            {
+                if (logTurnOnRed) {
+                    Log._Debug($"TurnOnRedManager.UpdateSegmentEnd({end.segmentId}, {end.startNode}): " +
+                               "right segment is not geometrically right");
+                }
+
+                rightSegmentId = 0;
+            }
+
+            // check for incoming one-ways
+            int indexL = segmentEndManager.GetIndex(leftSegmentId, nodeId);
```

Comment:
Creates false positive warning in log when `leftSegmentId` equals 0. Before change there was test for that case.

Reply:
Moved IndexL and IndexR back into the if ()
GetIndex() should not be logging anything, though, and should be returning a no-result value.